### PR TITLE
board: Adding support for W5500 Eval Board.

### PR DESCRIPTION
**Pull request by raveious**
The W5500_EVB_PICO is an evaluation board for the Wiznet W5500 ethernet mac/phy based on the Raspberry Pi Pico.

### CODE REVIEW

File: boards/arm/w5500_evb_pico/doc/img/w5500_evb_pico_side.png
Comment (nordicjm) [inline boards/arm/w5500_evb_pico/doc/img/w5500_evb_pico_side.png:1]: Resize this image to make it smaller then run it through https://tinypng.com/ or similar to get it to < 50KiB

**Review by nordicjm — DISMISSED** [review]

**Comment by raveious** [comment]
@anangl @soburi @yonsch Would you mind reviewing this?

**Review by soburi — APPROVED** [review]
LGTM

File: boards/arm/w5500_evb_pico/w5500_evb_pico.dts
Comment (nandojve) [inline boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: Is there any objections to have the serial console at USB like `wio_terminal` board?

Comment (nandojve) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: I just tested this patch + net/cloud/tagoio application and USB CDC + Ethernet works like a charm.

Comment (raveious) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: The original Raspberry Pi Pico board configuration has the console set to be uart0. Seeing as this board is essentially "a Pico with an ethernet jack glued to the side", I wasn't sure if I wanted stray far away from that.

Comment (raveious) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: Personally, I believe the board configs should be rather minimal sticking to what the physical hardware has, and bring in the least amount of code possible. This gives the application developer less work when they want to override functionality to remove what they aren't using.

Comment (nandojve) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: IMHO the drawback of not having the console in the USB is:

1- The user need to solder the header pins
2- Have an external USB<->Serial adapter
3- Maybe user change the code to enable CDC-ACM and have the console to avoid [1-2].

With the Serial Console in the USB by default it is easy to anyone validate the majority (if not all) examples and tests. Otherwise the user always have a lot of extra work with direct no benefit. Since the board came with 16MB of flash I don't see why not have it. If the user really need to use the serial port then he change what is needed because if their own use case for that.

The final decision is yours but I hope you could consider the change : )

Comment (raveious) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: Unless @anangl @yonsch @soburi or @nordicjm have a strong opinion on why it should be different than the original pico, I think I'm going to leave it as is. Sorry @nandojve .

Comment (soburi) [reply boards/arm/w5500_evb_pico/w5500_evb_pico.dts:24]: I agree it is good to keep this configuration the same as rpi_pico.

**Comment by raveious** [comment]
Some day this'll get a second approver, today is not that day.

**Review by yonsch — APPROVED** [review]

**Pull request merged**
